funnel/views/space.py

```diff
@@ -141,3 +141,23 @@ def space_edit(profile, space):
         flash(_("Your changes have been saved"), 'info')
         return redirect(space.url_for(), code=303)
     return render_form(form=form, title=_("Edit proposal space"), submit=_("Save changes"))
+
+
+@app.route('/<space>/rsvp', methods=['POST'], subdomain='<profile>')
+@lastuser.requires_login
+@load_models(
+    (Profile, {'name': 'profile'}, 'g.profile'),
+    ((ProposalSpace, ProposalSpaceRedirect), {'name': 'space', 'profile': 'profile'}, 'space'),
+    permission='view')
+def rsvp(profile, space):
+    if request.method == 'POST' and Form().validate_csrf_data(request.form['csrf_token']):
+        rsvp = Rsvp.query.get((space.id, g.user.id))
+        if rsvp:
+            rsvp.rsvp_action = request.form['rsvp_action']
```

Comment:
You're still not validating the value of `rsvp_action`. Just make a new form with one RadioField, with `choices=[(key, value['label']) for key, value in RSVP_ACTION.items()]`. Use it like this:

```python
form = RsvpForm()
if form.validate_on_submit():
    rsvp = Rsvp.get_for(space, g.user, create=True)
    form.populate_obj(rsvp)
    db.session.commit()
```

1. No need to make this code aware of Form internals like `validate_csrf_data` (it changes occasionally in WTForms without warning as it's meant to be internal detail).
2. Don't deal with ids like `space.id` outside the model itself. If the primary key order switches to `user_id, space_id` in future, you'll have to refactor across your code and errors won't show up easily because there's no validation on integers. We've had serious nightmares developing hacknight.in because of practices like this where the code worked as long as there was limited data in production and broke when users started adding data.